schedcat/locking/model.py

```diff
@@ -0,0 +1,48 @@
+import bounds
+
+class Spinlock(object):
+    def __init__(self, total_overhead=0, critical_overhead=0, apply_bounds_function=None):
+        if not apply_bounds_function:
+            apply_bounds_function = bounds.apply_pfp_lp_unordered_bounds
+        self._apply_bounds = apply_bounds_function
+        self.total_overhead = total_overhead
+        self.critical_overhead = critical_overhead
+
+    def apply_bounds(self, ts):
+        """Apply bounds and increase execution cost"""
+        self._apply_bounds(ts)
+        for _, t in enumerate(ts):
+            t.cost += t.blocked
```

Comment:
Note that this is incorrect in case of the LP-based analysis from the RTSS'13 paper—you shouldn't inflate execution costs. The analysis gives you a bound that should be directly used in the RTA as a blocking term.